gtsam/hybrid/HybridEliminationTree.h

```diff
@@ -0,0 +1,64 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file HybridEliminationTree.h
+ * @date Mar 11, 2022
+ * @author Fan Jiang
+ */
+
+#pragma once
+
+#include <gtsam/hybrid/HybridBayesNet.h>
+#include <gtsam/hybrid/HybridFactorGraph.h>
+#include <gtsam/inference/EliminationTree.h>
+
+namespace gtsam {
+
+/**
+ * Elimination Tree type for Hybrid
+ */
+class GTSAM_EXPORT HybridEliminationTree
+    : public EliminationTree<HybridBayesNet, HybridFactorGraph> {
+ public:
+  typedef EliminationTree<HybridBayesNet, HybridFactorGraph>
+      Base;                                    ///< Base class
+  typedef HybridEliminationTree This;          ///< This class
+  typedef boost::shared_ptr<This> shared_ptr;  ///< Shared pointer to this class
+
+  /**
+   * Build the elimination tree of a factor graph using pre-computed column
+   * structure.
+   * @param factorGraph The factor graph for which to build the elimination tree
+   * @param structure The set of factors involving each variable.  If this is
+   * not precomputed, you can call the Create(const FactorGraph<DERIVEDFACTOR>&)
+   * named constructor instead.
+   * @return The elimination tree
+   */
+  HybridEliminationTree(const HybridFactorGraph& factorGraph,
+                        const VariableIndex& structure, const Ordering& order);
+
+  /** Build the elimination tree of a factor graph.  Note that this has to
+   * compute the column structure as a VariableIndex, so if you already have
+   * this precomputed, use the other constructor instead.
+   * @param factorGraph The factor graph for which to build the elimination tree
+   */
+  HybridEliminationTree(const HybridFactorGraph& factorGraph,
+                        const Ordering& order);
+
+  /** Test whether the tree is equal to another */
+  bool equals(const This& other, double tol = 1e-9) const;
+
+ private:
+  friend class ::EliminationTreeTester;
```

Comment:
Interesting. Are we using GTest for this? I don't believe CppUnitLite needs friend class declarations.

Reply:
TBH I don't know, this is the case for all EliminationTrees